backend/sozisel/lib/sozisel_web/schema/resolvers/presenter_resolvers.ex

```diff
@@ -0,0 +1,89 @@
+defmodule SoziselWeb.Schema.Resolvers.PresenterResolvers do
+  alias SoziselWeb.Schema.Helpers
+  alias SoziselWeb.Schema.Subscriptions.Topics
+
+  alias Sozisel.Model.{
+    Sessions,
+    Events,
+    LaunchedEvents,
+    Quizzes
+  }
+
+  alias Events.Event
+  alias LaunchedEvents.LaunchedEvent
+  alias Sessions.Session
+  alias Quizzes.Quiz
+
+  def prepare_data_for_participants(
+        %Event{} = event,
+        %LaunchedEvent{} = launched_event
+      ) do
+    event_data = event.event_data
+
+    participant_event_data =
+      case event_data.__struct__ do
+        Quiz -> Events.prepare_quiz_data_for_participants(event_data)
+        _ -> {:error, "Wrong data"}
```

Comment:
It's event better to raise an exception here, we should not handle a situation where the __struct__ is unknwon to us. And I think that we could have `Events.marshall_participant_event_data(event_data)` and do the pattern matching of struct directly inside this function, something like this
```elixir
def marshal_participant_event_data(%{__struct__: Quiz} = event_data) do
...
```